### PR TITLE
[Task 1847362]4. add remaining logics to mysql-lib(create, get...)

### PR DESCRIPTION
**Pull request by andxu**
[AB#1847362](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1847362)

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-mysql-lib/src/main/java/com/microsoft/azure/toolkit/lib/mysql/service/MySqlManagerFactory.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    @Cacheable(cacheName = "MySqlManager", key = "$subscriptionId")
+    public static MySqlManager create(String subscriptionId) {
+        final Account account = Azure.az(AzureAccount.class).account();
+        final AzureConfiguration config = Azure.az().config();
+        final String userAgent = config.getUserAgent();
+        final HttpLogDetailLevel logLevel = Optional.ofNullable(config.getLogLevel()).map(HttpLogDetailLevel::valueOf).orElse(HttpLogDetailLevel.NONE);
+        final AzureProfile azureProfile = new AzureProfile(null, subscriptionId, account.getEnvironment());
+        return MySqlManager.configure()
+            .withLogOptions(new HttpLogOptions().setLogLevel(logLevel))
+            .withPolicy(getUserAgentPolicy(userAgent))
```
Comment (Flanker32) [inline azure-toolkit-libs/azure-toolkit-mysql-lib/src/main/java/com/microsoft/azure/toolkit/lib/mysql/service/MySqlManagerFactory.java:34]: may use `ProviderRegistrationPolicy` to register service provider automatically, refers https://github.com/microsoft/azure-maven-plugins/blob/develop/azure-toolkit-libs/azure-toolkit-applicationinsights-lib/src/main/java/com/microsoft/azure/toolkit/lib/applicationinsights/ApplicationInsights.java#L98

Comment (andxu) [reply azure-toolkit-libs/azure-toolkit-mysql-lib/src/main/java/com/microsoft/azure/toolkit/lib/mysql/service/MySqlManagerFactory.java:34]: will fix it in future PR

File: azure-toolkit-libs/azure-toolkit-mysql-lib/src/main/java/com/microsoft/azure/toolkit/lib/mysql/service/AzureMySql.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    public MySqlServer get(final String resourceGroup, final String name) {
+        final Server server = MySqlManagerFactory.create(getDefaultSubscription().getId()).servers().getByResourceGroup(resourceGroup, name);
+        return toMysqlServer(server);
+    }
+
+    public AbstractMySqlCreator create() {
+        final MySqlManager manager = MySqlManagerFactory.create(getDefaultSubscription().getId());
+        return new AbstractMySqlCreator() {
+
+            private MySqlServer app;
```
Comment (Flanker32) [inline azure-toolkit-libs/azure-toolkit-mysql-lib/src/main/java/com/microsoft/azure/toolkit/lib/mysql/service/AzureMySql.java:74]: app may be null before `commit`

File: azure-toolkit-libs/azure-toolkit-mysql-lib/src/main/java/com/microsoft/azure/toolkit/lib/mysql/service/AzureMySql.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+                    .withVersion(validateServerVersion(this.getVersion()));
+                Server server = manager.servers().define(getName())
+                    .withRegion(getRegion().getName())
+                    .withExistingResourceGroup(getResourceGroupName())
+                    .withProperties(parameters).create();
+                this.app = toMysqlServer(server);
+                return app;
+            }
+
+            @NotNull
```
Comment (wangmingliang-ms) [inline azure-toolkit-libs/azure-toolkit-mysql-lib/src/main/java/com/microsoft/azure/toolkit/lib/mysql/service/AzureMySql.java:91]: @Nonnull

File: azure-toolkit-libs/azure-toolkit-mysql-lib/src/main/java/com/microsoft/azure/toolkit/lib/mysql/service/AzureMySql.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+    private ServerVersion validateServerVersion(String version) {
+        if (StringUtils.isNotBlank(version)) {
+            final ServerVersion res = ServerVersion.fromString(version);
+            if (res == null) {
+                throw new AzureToolkitRuntimeException(String.format("Invalid mysql version '%s'.", version));
+            }
+            return res;
+        }
+        return null;
```
Comment (Flanker32) [inline azure-toolkit-libs/azure-toolkit-mysql-lib/src/main/java/com/microsoft/azure/toolkit/lib/mysql/service/AzureMySql.java:135]: Do we need to throw exception if version is empty?

File: azure-toolkit-libs/azure-toolkit-mysql-lib/src/main/java/com/microsoft/azure/toolkit/lib/mysql/service/AzureMySql.java [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+        return ServerVersion.values().stream().map(ExpandableStringEnum::toString).collect(Collectors.toList());
+    }
+
+    public boolean checkRegionAvailability(Region region) {
+        MySqlManager mySqlManager = MySqlManagerFactory.create(getDefaultSubscription().getId());
+        List<PerformanceTierProperties> tiers = mySqlManager.locationBasedPerformanceTiers().list(region.getName()).stream().collect(Collectors.toList());
+        return tiers.stream().anyMatch(e -> CollectionUtils.isNotEmpty(e.serviceLevelObjectives()));
+    }
+
+    private ServerVersion validateServerVersion(String version) {
```
Comment (Flanker32) [inline azure-toolkit-libs/azure-toolkit-mysql-lib/src/main/java/com/microsoft/azure/toolkit/lib/mysql/service/AzureMySql.java:127]: validate -> parse?

**Review by shenqianjin — APPROVED** [review]
Approve as discuss in the meeting

**Pull request merged**